src/uwtools/drivers/forecast.py

```diff
@@ -121,15 +129,92 @@ def requirements(self) -> None:
         ???
         """
 
-    def resources(self) -> None:
+    def resources(self, platform: dict) -> Mapping:
         """
-        ???
+        Set up the formatting for the scheduler batch script.
         """
-
-    def run(self):
+        # Add required fields to platform.
+        # Currently supporting only slurm and none.
+        slurmqueues = ["hera", "jet", "orion", "stampede"]
+        if platform["queue"] is None and platform["machine"] in slurmqueues:
+            platform["queue"] = "slurm"
+        else:
+            platform["queue"] = "none"
+
+        return {
+            "account": platform["account"],
+            "nodes": 1,
+            "queue": platform["queue"],
+            "scheduler": platform["scheduler"],
+            "tasks_per_node": 1,
+            "walltime": "00:01:00",
+        }
+
+    def run(self, log: Optional[Logger] = None) -> None:
         """
         Runs FV3.
         """
+        # Set up logging.
+        if log is None:
+            name = f"{inspect.stack()[0][3]}"
+            log = cli_helpers.setup_logging(
+                log_file="/dev/stdout",
+                log_name=name,
+                quiet=False,
+                verbose=False,
+            )
+        # Read in the config file.
+        forecast_config = config.YAMLConfig(self._config_file)
+
+        # Define forecast model.
+        forecast_model = forecast_config["forecast"]["EXTRN_MDL_NAME"]
+        machine = forecast_config["platform"]["machine"]
+
+        # Prepare directories.
+        run_directory = forecast_config["forecast"]["FCSTDIR"]
+        self.create_directory_structure(run_directory, "delete")
+
+        static_files = forecast_config["forecast"]["STATIC"]
+        self.stage_files(run_directory, static_files, link_files=False)
+        cycledep_files = forecast_config["forecast"]["CYCLEDEP"]
+        self.stage_files(run_directory, cycledep_files, link_files=True)
```

Comment:
It's likely safer to link the static files, and copy the cycledep ones by default.

Reply:
ok, we might want to recheck how stage_files() is set up here and in test_forecast() too then

Reply:
I think the `stage_files` method doesn't care which way it goes, so any combination would be fine. This request is because static files are typically staged on disk and don't often move or change (although it can happen), while many times your cycle-dependent files will be scrubbed during the course of your experiment. It's safer to keep those big files with your experiment.

Reply:
I'm not suggesting a change now, but just throwing this out for maybe future consideration: Making a hardlink when possible (that is, when source and destination files are on the same filesystem) would be an ideal first thing to try, falling back to a copy if that fails: A hardlink uses the same underlying blocks on disk, and therefore 1. is instantaneous and 2. uses no extra disk space. If the "source" is removed, the "destination" remains. Hardlinks will fail either if the source is on a different filesystem, or if the user doesn't have write access to the source. But for cases where those requirements are met, it can save a lot of time and space. For example, if a user has their own collection of files and is running many experiments against them, hardlinks can avoid eating in to their quota.

Reply:
I've moved both to link for now after today's discussion